azurelinuxagent/common/protocol/wire.py

```diff
@@ -779,17 +779,17 @@ def update_goal_state(self, forced=False, max_retry=3):
 
                 return
 
+            except IOError as e:
+                logger.warn("IOError processing goal state, retrying [{0}]", ustr(e))
+
             except ResourceGoneError:
                 logger.info("GoalState is stale -- re-fetching")
                 goal_state = None
 
             except Exception as e:
-                log_method = logger.info \
-                                if type(e) is ProtocolError \
-                                else logger.warn
-                log_method(
-                    "Exception processing GoalState-related files: {0}".format(
-                        ustr(e)))
+                log_method = logger.verbose if type(e) is ProtocolError else logger.error
```

Comment:
I'm not a big fan of this idiom (catch Exception, then change behavior based on checking type()). I'd rather see two separate "except" statements which are explicit about what they do.
```
except ProtocolError as e:
    logger.verbose(exception_msg.format(ustr(e), traceback.format_exc()))
except Exception as e:
    logger.error(exception_msg.format(ustr(e), traceback.format_exc()))
```